handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java

```diff
@@ -160,9 +199,11 @@
 
     private volatile ClientAuth clientAuth = ClientAuth.NONE;
 
-    private volatile String endPointIdentificationAlgorithm;
+    private String endPointIdentificationAlgorithm;
     // Store as object as AlgorithmConstraints only exists since java 7.
-    private volatile Object algorithmConstraints;
+    private Object algorithmConstraints;
+    @SuppressWarnings("rawtypes")
+    private volatile List sniHostNames;
```

Comment:
it looks like this is only used in `synchronized` blocks ... does it need to be volatile too?